openmc/geometry.py

```diff
@@ -151,123 +151,98 @@ def get_cell_instance(self, path):
-        universes.sort(key=lambda x: x.id)
+        universes = OrderedDict()
+        universes[self.root_universe.id] = self.root_universe
+        universes.update(self.root_universe.get_all_universes())
         return universes
 
     def get_all_materials(self):
-        """Return all materials assigned to a cell
+        """Return all materials within the geometry.
 
         Returns
         -------
-        list of openmc.Material
-            Materials in the geometry
+        collections.OrderedDict
+            Dictionary mapping material IDs to :class:`openmc.Material`
+            instances
 
         """
-
-        material_cells = self.get_all_material_cells()
-        materials = []
-
-        for cell in material_cells:
-            if cell.fill_type == 'distribmat':
-                for m in cell.fill:
-                    if m is not None and m not in materials:
-                        materials.append(m)
-            elif cell.fill_type == 'material':
-                if cell.fill not in materials:
-                    materials.append(cell.fill)
-
-        materials.sort(key=lambda x: x.id)
-        return materials
+        return self.root_universe.get_all_materials()
 
     def get_all_material_cells(self):
         """Return all cells filled by a material
 
         Returns
         -------
-        list of openmc.Cell
-            Cells filled by Materials in the geometry
+        collections.OrderedDict
+            Dictionary mapping cell IDs to :class:`openmc.Cell` instances that
+            are filled with materials or distributed materials.
 
         """
+        material_cells = OrderedDict()
 
-        all_cells = self.get_all_cells()
-        material_cells = []
-
-        for cell in all_cells:
+        for cell in self.get_all_cells().values():
             if cell.fill_type in ('material', 'distribmat'):
                 if cell not in material_cells:
-                    material_cells.append(cell)
+                    material_cells[cell.id] = cell
 
-        material_cells.sort(key=lambda x: x.id)
         return material_cells
 
     def get_all_material_universes(self):
-        """Return all universes composed of at least one non-fill cell
+        """Return all universes having at least one non-fill cell
 
         Returns
         -------
-        list of openmc.Universe
-            Universes with non-fill cells
+        collections.OrderedDict
+            Dictionary mapping universe IDs to :class:`openmc.Universe`
+            instances with non-fill cells
 
         """
+        material_universes = OrderedDict()
 
-        all_universes = self.get_all_universes()
-        material_universes = []
-
-        for universe in all_universes:
-            cells = universe.cells
-            for cell in cells:
+        for universe in self.get_all_universes():
+            for cell in universe.cells:
                 if cell.fill_type in ('material', 'distribmat', 'void'):
```

Comment:
It might be nice to clarify in the docstring that this method treats `'void'` as a "material".